subsys/net/lib/nrf_cloud/src/nrf_cloud_credentials.c

```diff
@@ -192,27 +241,50 @@ int nrf_cloud_credentials_check(struct nrf_cloud_credentials_status *const cs)
 
 	ret = cred_exists(cs->sec_tag, CA_CERT, &exists);
 	if (ret < 0) {
+		LOG_ERR("Error checking CA exists");
 		return -EIO;
 	}
 	cs->ca = exists;
+	if (exists) {
+		ret = cred_size_get(cs->sec_tag, CA_CERT, &cs->ca_size);
+		if (ret < 0) {
+			LOG_ERR("Error checking CA size");
+			return -EIO;
+		}
+		/* These flags are approximate and only useful for logging to help diagnose
+		 * possible provisioning mistakes.
+		 */
+		cs->ca_combined = (cs->ca_size > CONFIG_NRF_CLOUD_COMBINED_CA_CERT_SIZE_THRESHOLD);
+		cs->ca_aws = !cs->ca_combined &&
+			     (cs->ca_size > CONFIG_NRF_CLOUD_COAP_CA_CERT_SIZE_THRESHOLD);
+		cs->ca_coap = !cs->ca_aws && !cs->ca_combined &&
+			      (cs->ca_size > (CONFIG_NRF_CLOUD_COAP_CA_CERT_SIZE_THRESHOLD / 2));
+	}
 
 	ret = cred_exists(cs->sec_tag, CLIENT_CERT, &exists);
 	if (ret < 0) {
+		LOG_ERR("Error checking client cert exists");
 		return -EIO;
 	}
 	cs->client_cert = exists;
 
 	ret = cred_exists(cs->sec_tag, PRIVATE_KEY, &exists);
 	if (ret < 0) {
+		LOG_ERR("Error checking private key exists");
 		return -EIO;
 	}
 	cs->prv_key = exists;
 
-	LOG_DBG("Sec Tag: %u, CA: %s, Client Cert: %s, Private Key: %s",
+	LOG_DBG("Sec Tag: %u; CA: %s, Client Cert: %s, Private Key: %s",
 		cs->sec_tag,
 		cs->ca		? "Yes" : "No",
 		cs->client_cert	? "Yes" : "No",
 		cs->prv_key	? "Yes" : "No");
+	LOG_DBG("CA Size: %zd, Combined: %s, AWS only: %s, CoAP only: %s",
+		cs->ca_size,
+		cs->ca_combined ? "Yes" : "No",
+		cs->ca_aws ? "Yes" : "No",
+		cs->ca_coap ? "Yes" : "No");
```

Comment:
Instead of yes/no, perhaps use `likely/unlikely`?

We wouldn't want a user to trust these indications implicitly and be misled